Sources/TuistGenerator/Linter/EnvironmentLinter.swift

```diff
@@ -36,18 +36,14 @@ public class EnvironmentLinter: EnvironmentLinting {
     /// - Returns: An array with a linting issue if the selected version is not compatible.
     /// - Throws: An error if there's an error obtaining the selected Xcode version.
     func lintXcodeVersion(config: Config) throws -> [LintingIssue] {
-        guard case let CompatibleXcodeVersions.list(compatibleVersions) = config.compatibleXcodeVersions else {
-            return []
-        }
-
         guard let xcode = try XcodeController.shared.selected() else {
             return []
         }
 
         let version = xcode.infoPlist.version
 
-        if !compatibleVersions.contains(version) {
-            let versions = compatibleVersions.joined(separator: ", ")
+        if !config.compatibleXcodeVersions.isCompatible(versionString: version) {
+            let versions = config.compatibleXcodeVersions
             let message =
                 "The project, which only supports the versions of Xcode \(versions), is not compatible with your selected version of Xcode, \(version)"
```

Comment:
If were in here, would be sweet to have the error be something like:

> The selected Xcode version is x.x.x which is not compatible with this projects Xcode version requirement of 1.0.0 up to 2.0.0

> The selected Xcode version is x.x.x which is not compatible with this projects Xcode version requirement of 1.0.0

> The selected Xcode version is x.x.x which is not compatible with this projects Xcode version requirements of 1.0.0, 2.1.0, 3.0.0

Reply:
Sure, the error message can be improved.

I don't think saying "up to 2.0.0" (for `.upToNextMajor("1.0")`) is a good idea, as that could be misinterpreted to mean 2.0.0 is valid. And, of course, we don't know what the last version before the next major will be. I'll see what I can come up with.

Reply:
That works for me, and yeah I noticed the possibility of misinterpretation there with the plain English. You could also just display it in swift range format? Ex: `x.x.x..<x.x.y` which Swift developers may be more familiar with (and I think SPM does it like this)